cli/deepsea.py

```diff
@@ -119,80 +122,144 @@ def _validate_stage_file_exists(stage_name):
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_success_deps]
+                deps.sort()
+                PP.print(PP.dark_yellow("if_success=[{}] ".format(", ".join(deps))))
+            if step.on_fail_deps:
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_fail_deps]
+                deps.sort()
+                PP.println(PP.dark_yellow("if_fail=[{}]".format(", ".join(deps))))
+            else:
+                PP.println()
+
+    elif isinstance(step, (SaltStateFunction, SaltExecutionFunction)):
+        PP.print(PP.grey(step.function))
+        if step.args:
+            args_str = ", ".join(step.args)
+            if len(args_str) > 60:
+                args_str = args_str.replace("\n", "\\n")
+                args_str = args_str[:57] + "..."
+            PP.print(PP.grey("({})".format(args_str)))
+        if indent == 10:
+            PP.print(PP.dark_green(" ({})".format(step.desc)))
+            if step.target:
+                PP.print(PP.grey(" on "))
+                PP.print(PP.cyan(", ".join(step.target)))
+        PP.println()
+        if step.on_success_deps or step.on_fail_deps:
+            if step.on_success_deps:
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_success_deps]
+                deps.sort()
+                PP.print(PP.dark_yellow("if_success=[{}] ".format(", ".join(deps))))
+            if step.on_fail_deps:
+                PP.print("{}".format(" " * indent))
+                deps = [step_order_map[id(s)] for s in step.on_fail_deps]
+                deps.sort()
+                PP.println(PP.dark_yellow("if_fail=[{}]".format(", ".join(deps))))
+            else:
+                PP.println()
+
+
+def _run_show_stage_steps(stage_name, hide_state_steps, only_visible_steps):
     """
     Runs stage parser and prints the list of steps
     """
     _validate_stage_file_exists(stage_name)
 
-    PP.p_header("Parsing stage: {}".format(stage_name))
-    steps, _ = SLSParser.parse_state_steps(stage_name, hide_state_steps, only_visible_steps,
-                                           use_cache)
-    print()
-    PP.p_bold("List of steps for stage {}:".format(stage_name))
-    print()
+    PP.print(PP.dark_yellow("Parsing {} steps... ".format(stage_name)))
+    PP.flush()
+    t0 = time.time()
+
+    try:
+        steps, _ = SLSParser.parse_stage(stage_name, hide_state_steps,
+                                         only_visible_steps)
+    except StageRenderingException as ex:
+        PP.println(PF.FAIL)
+        PP.println()
+        PP.println(PP.bold("An error occurred while rendering stage: {}"
+                           .format(ex.stage_name)))
+        for error in ex.error_list:
+            PP.println(PP.red("  - {}".format(error)))
+        PP.println()
+        return
```

Comment:
I think it is better to have an explicit `return None` for readability

Reply:
but this function does not return anything (void)